Unity-Technologies/Graphics
# Added an hybrid RayMarching/RayTracing mode for RTReflections and RTGI.

This PR Adds a new control mode for ray traced reflections and ray traced global illumination. This new mode combines both ray marching and ray tracing to solve visibility. This allows for both potiential peformance boost and access to details that cannot be stored in the RTAS (Decals or opaque particles). The option can be found on both volumes and are only compatible with the performance mode.

![image](https://user-images.githubusercontent.com/42026998/109839202-2f1ec500-7c47-11eb-8088-f7ba17a723b1.png)

**Testing status**
Tested locally extensively with multiple scenes and assets.
Addeds 4 new tests to cover this new features.
QA needs to test because this changes quite a bit of code and is complex. Here are some things to test
- Verify that hybrid and ray tracing is the same (minus artefacts)
- Reflect sky should work the same way in both modes
- Test that supported ray tracing modes (in Asset) works as expected (Hybrid is supported only in both or perf mode)
- Test that hybrid mode works correctly on transparent that receives RTR
- Ensure that both parameters ray steps works as expected
- Test half resolution in hybrid
- Test Vertex Animation in hybrid
